Feature/guild portal page

0xvangrim opened the pull request:
# Description
[Guilds landing Page](https://github.com/DXgovernance/dxvote/issues/678)
<img width="770" alt="Screenshot 2022-03-20 at 22 24 06" src="https://user-images.githubusercontent.com/59825171/159186729-5005c774-ee3b-432b-84ca-ba0f4f9f9914.png">

The first implementation of the Guilds Landing Page together with the new SC `GuildRegistry`. Except for the design, these are the following features enabled:

- Dynamic guilds amount that fetches data from the SC
- Dynamic names (ENS)
- Dynamic subdomains (ENS)

TODO (Not sure if any of the below can be down as of now?)
- Add dynamic member count
- Add dynamic proposal count
- Add dynamic logo

The subdomain name is simply pushed to the back looking like `guilds/rinkeby/[RANDOM_GUILD]`. I am aware that we are going to change it later to be a proper subdomain. Should be an easy fix when we've deployed the new ENS name though.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [X ] My code follows the existing style of this project
- [X ] I have performed a self-review of my own code
- [X ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ X ] Any UI changes have been tested and made responsive for mobile views

Commit:
made the portal page with hardcoded data

Commit:
format

Commit:
format

Commit:
format

rossneilson committed:
Merge pull request #693 from DXgovernance/develop

Version 1.0.11

Commit:
guild amount dynamic, subdomains dynamic, names dynamic

Commit:
remove comment

Commit:
fix conflict

Commit:
added question marks if name doesn't exist

Commit:
format

carlbarrdahl commented on src/pages/Guilds/LandingPage.tsx:
```diff
@@ -0,0 +1,132 @@
+        />
+        <StyledButton
+          variant="secondary"
+          onClick={() => history.push(location.pathname + '/createGuild')}
```
I would prefer to use React Routers Link component instead because onClick links doesn't have the capability of shift click to open new tabs or right click.

carlbarrdahl commented on src/pages/Guilds/LandingPage.tsx:
```diff
@@ -0,0 +1,132 @@
+      <CardContainer>
+        {totalGuilds
+          ? totalGuilds.map((guildAddress, key) => (
+              <GuildCard key={key} guildAddress={guildAddress}>
```
Using the array index as key might cause problems if the order of the `totalGuilds` changes (for example with sorting or filtering). The `guildAddress` can be used instead.

More info here: https://reactjs.org/docs/lists-and-keys.html#keys

carlbarrdahl commented on src/hooks/Guilds/ether-swr/guild/useGuildRegistry.ts:
```diff
@@ -0,0 +1,19 @@
+  return {
+    totalGuilds,
+    error,
+  };
```
I prefer to always return `{ data, loading, error }` and keep it consistent between hooks. By following this pattern we know that a data fetching hook will always have these props to be used by the component calling it.

The component can always rename these props if there are multiple hooks:
```js
function GuildsList({ contractAddress }) {
  // with deconstruction
  const { data: totalGuilds, loading: loadingGuilds, error: errorGuilds } = useGuildRegistry(contractAddress) 
  // or as object
  const otherHook = useAnotherHook() // otherHook.data | loading | error
```

carlbarrdahl reviewed (commented):
Looks great! Added a few comments about some small details.

0xvangrim commented:
Thanks for the review @carlbarrdahl! Will take a look at it!

hhamud commented on src/components/Guilds/common/Form/Input/Input.tsx:
```diff
@@ -7,8 +7,11 @@ const baseInputStyles = css`
-    color: ${({ theme }) => theme.colors.proposalText.lightGrey};
+    color: ${({ theme }) => theme.colors.proposalText.grey};
+    font-size: 14;
+    font-weight: 500;
```
Should use either of the font variants in themes

hhamud commented on src/pages/Guilds/LandingPage.tsx:
```diff
@@ -0,0 +1,132 @@
+import useENSNameFromAddress from 'hooks/Guilds/ether-swr/ens/useENSNameFromAddress';
+
+const InputContainer = styled(Flex)`
+  display: flex;
```
no need for display flex when using Flex

hhamud commented on src/pages/Guilds/LandingPage.tsx:
```diff
@@ -0,0 +1,132 @@
+          placeholder="Search Guild"
+        />
+        <StyledButton
+          variant="secondary"
```
remove the inline variant and replace it with this
```suggestion
         const StyledButton = styled(Button).attrs(() => ({
         variant: 'secondary",
         })`
        margin-left: 1rem;
        width: 9rem;
        padding: 0.7rem;
        color: ${({ theme }) => theme.colors.text};
`;

```

hhamud commented on src/hooks/Guilds/ether-swr/guild/useGuildRegistry.ts:
```diff
@@ -0,0 +1,19 @@
+import GuildRegistry from 'contracts/GuildsRegistry.json';
+
+export const useGuildRegistry = (contractAddress: string) => {
+  const { data: totalGuilds, error } = useEtherSWR(
```
I think we should memoize this

0xvangrim replied:
Yeah, I thought about it too. But this is what I get when I try to do it: 
![image](https://user-images.githubusercontent.com/59825171/159951080-48287fe4-8423-49ef-b261-1804744846ab.png)

Since the hook needs to be called either way and we're not doing an expensive function call I thought it would be fine to leave it as it is. Or do you have another idea?

Commit:
fixed comments

Commit:
fixed conflicts

0xvangrim commented:
@carlbarrdahl @hhamud 
I've now addressed most of the issues above:
- Changed from using the hook`useHistory` to `Link`
- Refactored a bit on the styling structure
- Changed naming to be consistent in the hooks

Thanks again! :)

Commit:
fix format

Commit:
replaced history with Link

Commit:
format again.
